lib/before-prepareJS.js

```diff
@@ -10,6 +10,6 @@ module.exports = function ($projectData, $logger, hookArgs) {
         bundle: appFilesUpdaterOptions.bundle,
         release: appFilesUpdaterOptions.release,
     };
-    const result = config.bundle && runWebpackCompiler.bind(runWebpackCompiler, config, $projectData, $logger, hookArgs);
+    const result = config.bundle && runWebpackCompiler.bind(runWebpackCompiler, config, hookArgs.config.projectData, $logger, hookArgs);
```

Comment:
I guess this (`hookArgs.config...`) is intentionally different from the rest but I couldn't sort it out why by looking briefly through the cli source.

Reply:
The hookArgs object contains the arguments passed to the decorated method. The current hook is executed when the preparePlatform method in CLI is called: https://github.com/NativeScript/nativescript-cli/blob/97d72e9f414b1911982af715b36ae21f599cf32c/lib/services/prepare-platform-js-service.ts#L36
As you can see, this method accepts a single argument called `config`. So the `hookArgs` object in the current context will have a single property - `config` and its value will be the value passed to the `preparePlatform` method. We know this value will be object which has projectData property.

For the other hooks the projectData is often a direct argument of the decorated method, for example: https://github.com/NativeScript/nativescript-cli/blob/master/lib/services/platform-service.ts#L283-L287